cu_servicenow/cu_servicenow.module

```diff
@@ -0,0 +1,316 @@
+    $form['field_cu_servicenow']['#access'] = FALSE;
+    dpm($form['field_cu_servicenow']);
+  }
+}
+
+function servicenow_incident_mapping_submit($form, &$form_state) {
+  $nid = $form['#node']->nid;
+  $sys_id = $form_state['values']['options'];
+  $incidents = _servicenow_incident_assignments();
+
+  $sn_full_incident_info = [];
+  foreach ($incidents->result as $incident) {
+    $sn_full_incident_info[$incident->sys_id] = ['nature_of_issue' => $incident->u_nature_of_issue, 'business_service' => $incident->u_business_service];
+  }
+
+  $output = [
+    'sys_id' => $sys_id,
+    'nature_of_issue' => $sn_full_incident_info[$sys_id]['nature_of_issue'],
+    'business_service' => $sn_full_incident_info[$sys_id]['business_service'],
+  ];
+
+  variable_set('sn_incident_' . $nid, $output);
+  variable_set('sn_incident_sys_id_' . $nid, $output['sys_id']);
+  watchdog('cu_servicenow', t('Incident assignment has been made for NID: :nid', [':nid' => $nid]));
+}
+
+function servicenow_save_field_map($form, &$form_state) {
+  $current_vars = variable_get('servicenow_field_mapping', []);
+  if (isset($form_state['values']['servicenow']['field_map'])) {
+    $sn_value = [$form_state['values']['form_key'] => $form_state['values']['servicenow']['field_map']];
+    $sn_value = array_merge($current_vars, $sn_value);
+    variable_set('servicenow_field_mapping', $sn_value);
+    drupal_set_message(t('Successfully mapped webform field to ServiceNow.'), 'notice');
+  }
+}
+
+/**
+ * Custom submit handler for forms with SN integaration enabled.
+ */
+function servicenow_submit(&$form, &$form_state) {
+  global $user;
+  $nid = $form['#node']->nid;
+
+  // If the current node has SN enabled, do code, else move on.
+  watchdog('cu_servicenow', 'ServiceNow custom submit handler was hit.');
+  $form_values = $form_state['input']['submitted'];
+  $servicenow_field_map = variable_get('servicenow_field_mapping', '');
+
+  foreach($form_values as $key => $value) {
+    if(key_exists($key, $servicenow_field_map)) {
+      $sn[$servicenow_field_map[$key]] = $value;
+    }
+  }
+
+  $sn_incident_map = variable_get('sn_incident_' . $nid, []);
+
+  $sn['task_for'] = $user->name;
+  $sn['u_alternate_email'] = $user->mail;
+  $sn['state'] = '9';
+  $sn['contact_type'] = 'self-service';
+  $sn['cmdb_ci'] = $sn_incident_map['business_service']; // Business Service
+  $sn['u_nature_of_issue'] = $sn_incident_map['nature_of_issue']; // Nature of Issue
+  $sn['impact'] = '9c6238b555beb00096fffc196a1bb892'; // Impact
+  $sn['urgency'] = '9c6238b555beb00096fffc196a1bb892'; // Urgency
+
+  // @todo add incident assignments from form options/settings.
+
+  _servicenow_curl_request($sn);
+  //dpm($sn);
+
+  // Test uploading files
+  //_servicenow_upload_file($_FILES['files']['tmp_name']['submitted_attachment'], $_FILES['files']['type']['submitted_attachment'], $_FILES['files']['name']['submitted_attachment']);
+}
+
+function _servicenow_incident_assignments() {
+  global $conf;
```

Comment:
variable_get() uses the $conf array, so you could use that function to provide defaults if the variable doesn't exist or some check if the variable is missing.